Source/Core/Common/FileUtil.cpp

```diff
@@ -464,14 +464,12 @@ FSTEntry ScanDirectoryTree(const std::string& directory, bool recursive)
   {
     const std::string virtual_name(TStrToUTF8(ffd.cFileName));
 #else
-  struct dirent dirent, *result = nullptr;
-
   DIR* dirp = opendir(directory.c_str());
   if (!dirp)
     return parent_entry;
 
   // non Windows loop
-  while (!readdir_r(dirp, &dirent, &result) && result)
+  while (struct dirent* result = readdir(dirp))
```

Comment:
You only need struct in C. In C++, you don't need struct unless it's ambiguous. (@lioncash pointed this out to me before)